Minor doc fixes

basil-conto opened the pull request:
Commits descriptions:

1. The main part of this PR, fixing actual doc errors/ambiguities.
   The rest of the commits are merely stylistic suggestions.
2. Various suggestions for minor rewording.
3. Attempt to format metasyntactic variables like `(FOO . BAR)` as code, like the built-in Elisp manual does (I haven't exported the Org source so I'm not sure my patch actually works). To see what I mean, compare the formatting in the Ghub Info manual:

   ![ghub](https://user-images.githubusercontent.com/9121222/40328561-af8eab9a-5d3e-11e8-87a7-b296480cf1a2.png)
   with that in the Elisp Info manual:

   ![elisp](https://user-images.githubusercontent.com/9121222/40328622-e2a3c51a-5d3e-11e8-932c-a54dc2a778a4.png)
4. Locally enforce currently implied fill variables in the manual.
5. Consistently write acronyms in upper case (unsure about those `LocalWords` settings).
6. Enable URL hyperlinking in docstrings as per [`(elisp) Documentation Tips`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Documentation-Tips.html).

tarsius commented on ghub.org:
```diff
@@ -90,7 +90,7 @@ of sharing a single token between all packages that rely on it.  That
 new package without any additional setup.  But Ghub helps with that.
 
-When the user invokes some command defined in some package and that
+When the user invokes some command defined in some package which
```
I believe one ought to use `that` when the meaning is lost after removing `that/[,]which ...`. So I guess just drop the `and`?

npostavs replied:
Maybe drop `defined in some package` too? I think it just makes the sentence more confusing.

tarsius replied:
Agreed.

tarsius commented:
> Consistently upcase acronyms 

Where they consistently lowercase before? I think that was intentional to avoid acronyms being confused with ARGUMENTS.

tarsius commented:
> at URL `https://magit.vc/goto/2e586d36'

Lets do that, even though it's a bit silly that this is required and despite it also suffering from the ARGUMENT-or-ACRONYM? issue.

tarsius commented:
Actually, since we do the latter, we might as well upcase all the other acronyms too.

Which only leaves the *that vs. which* issue. What do you think about that?

tarsius commented:
In the manual we could remove the ambiguity by quoting

```
Optional ~ARGUMENT~, if non-nil, is a JSON-like list.
```

basil-conto commented:
> > Consistently upcase acronyms

> Where they consistently lowercase before?

Not in the manual, e.g. the documentation for `ghub-graphql` refers to upper case JSON.

> I think that was intentional to avoid acronyms being confused with ARGUMENTS.

Makes sense, though I'm in favour of consistency either way, and IME Emacs' help system is smart enough to determine which upper case words are arguments. Note, for example, how `JSON` and `STRING` are formatted differently in the docstring of `json-parse-string`:

![json](https://user-images.githubusercontent.com/9121222/40333691-b97dda00-5d51-11e8-99a0-540e1427debf.png)

> Which only leaves the `that vs. which` issue. What do you think about that?

I agree with you that the only "issue" is the presence of the "and" ("there" vs "which" comes down to style here, AFAIK), and I agree with @npostavs's suggestion to further simplify the sentence.

> In the manual we could remove the ambiguity by quoting

Good point. Updated patches to follow.

basil-conto commented:
> > In the manual we could remove the ambiguity by quoting

> Good point. Updated patches to follow.

That would, however, be breaking convention with the Emacs/Elisp manuals. Is that a problem?

basil-conto commented:
> Updated patches to follow.

I've added a few more typos to the first commit, resolved *that vs which* as suggested, and added an extra commit marking all metasyntactic variables up as code. This way you can just omit the final commit if you decide against this convention.

tarsius commented:
Thanks again! I completely forgot about this pr.

tarsius commented:
When regenerating the manual after your changes stray `@` appear in random places, but they seem to have no effect. See https://github.com/magit/ghub/tree/stray-at. Any idea why that might be?

tarsius commented:
I have temporarily removed your commits again until we have figured out what is going on here.

tarsius commented:
Actually that happens even without your changes. The `@` always appears before the `.` ending a sentences, but only in some sentences and I don't see what is special about those sentences.

tarsius commented:
This is due to https://github.com/emacsmirror/org/commit/92b03d161ab16e7352bef4980a8f19e00011e14f, but I don't really understand the change or the commit message.

tarsius commented:
In summary, I have merged your changes and regenerated the manual. :wink:

basil-conto commented:
Thanks!
